ibis/backends/base/sqlglot/rewrites.py

```diff
@@ -157,7 +157,9 @@ def extract_ctes(node):
 
     g = Graph.from_bfs(node, filter=(ops.Relation, ops.Subquery, ops.JoinLink))
     for node, dependents in g.invert().items():
-        if len(dependents) > 1 and isinstance(node, cte_types):
+        if isinstance(node, ops.View) or (
```

Comment:
Could we avoid having this special case here? IIC it is only to force turning a `View` into a CTE.

Without this the generated SQL would be a subselect, right?

Reply:
Without this change a cute won't be created, and View _must_ compile to a cte so it's visible by any subsequent query.

Reply:
Consider the case of joining two aliased ibis expressions. The aliases need to be visible at the top level for a subsequent join that uses them. An aliased subselect would hide all aliases except the immediate parent. AFAIK only CTEs allow defining multiple named relations with the same scope.

Reply:
This code still confuses me, you add `View` here to the list of the relations which should be wrapped with a CTE, but you exclude the same type from the dict comprehension below.

Reply:
I'm adding it to the list to be able to generate CTEs later, the ibis `CTE` object is only used in a call to `replace`, the dict of which filters out the `View` objects. There's are a few separate components of that process, that are _independent of this PR_:

1. The extraction of things that should become CTEs
2. The replacement of existing nodes with those CTEs
3. The construction of sqlglot CTEs based on step 1

I am reusing steps 1 and 3 for this PR, but operating on `View`s.

Reply:
The list needs to be assembled in topological order, we can't assemble the View CTEs separately from the others.